packages/@aws-cdk/pipelines/lib/legacy/stage.ts

```diff
@@ -225,6 +290,61 @@ export class CdkStage extends CoreConstruct {
     return stripPrefix(s, `${this.stageName}-`);
   }
 
+  /**
+   * Add a security check before the prepare/deploy actions of an CDK stage.
+   * The security check consists of two actions:
+   *  - CodeBuild Action to check for security changes in a stage
+   *  - Manual Approval Action that is auto approved via a Lambda if no security changes detected
+   */
+  private addSecurityCheck(appStageName: string, options?: BaseStageOptions) {
+    const { cdkDiffProject } = this.getApplicationSecurityCheck();
+    const notificationTopic: sns.ITopic | undefined = options?.securityNotificationTopic ?? this.securityNotificationTopic;
+    notificationTopic?.grantPublish(cdkDiffProject);
+
+    const approveActionName = `${appStageName}ManualApproval`;
+    const diffAction = new CodeBuildAction({
+      runOrder: this.nextSequentialRunOrder(),
+      actionName: `${appStageName}SecurityCheck`,
+      input: this.cloudAssemblyArtifact,
+      project: cdkDiffProject,
+      variablesNamespace: `${appStageName}SecurityCheck`,
+      environmentVariables: {
+        STACK_NAME: {
```

Comment:
I'm noticing we have `STACK_NAME` here -- what if there are more than one stacks in the stage we're deploying?

Reply:
Ah so this is the pipeline stack name..not like a stack within the stage